src/Microsoft.AspNetCore.Razor.Tools/GenerateCommand.cs

```diff
@@ -97,16 +119,26 @@ private int ExecuteCore(string projectDirectory, string outputDirectory, string
                     }
                 }
 
-                var viewFile = result.ViewFileInfo.ViewEnginePath.Substring(1);
-                var outputFileName = Path.ChangeExtension(viewFile, ".cs");
-
-                var outputFilePath = Path.Combine(outputDirectory, outputFileName);
+                var outputFilePath = result.InputItem.OutputPath;
                 File.WriteAllText(outputFilePath, result.CSharpDocument.GeneratedCode);
             }
 
             return success ? 0 : -1;
         }
 
+        private MemoryRazorProject GetMemoryProject(InputItem[] inputItems)
+        {
+            var project = new MemoryRazorProject();
+            foreach (var item in inputItems)
+            {
+                var projectItem = MemoryRazorProject.CreateProjectItem(item.CreateReadStream, item.PhysicalRelativePath, item.SourcePath, item.FilePath);
```

Comment:
Could just simplify InputItem to have a `RazorProjectItem` member. It basicccally has all the same information except for the `OutputPath`